browser/day_zero_browser_ui_expt/day_zero_browser_ui_expt_manager.cc

```diff
@@ -32,17 +40,34 @@ DayZeroBrowserUIExptManager::Create(ProfileManager* profile_manager) {
   }
 
   // This class should be instantiated after getting valid first run time;
-  if (brave_stats::GetFirstRunTime(nullptr).is_null()) {
+  if (brave_stats::GetFirstRunTime(g_browser_process->local_state())
+          .is_null()) {
     // This should not be happened in production but not 100% in the wild(ex,
     // corrupted user data). Just early return for safe. If upstream changes the
     // timing of fetching first run time, browser test will catch this.
     LOG(ERROR) << __func__ << " This should be happened only on test.";
     return nullptr;
   }
 
+  std::optional<std::string> day_zero_variant;
+  if (base::FeatureList::IsEnabled(features::kBraveDayZeroExperiment)) {
+    day_zero_variant = features::kBraveDayZeroExperimentVariant.Get();
+  }
+
+  if (!day_zero_variant) {
+    LOG(ERROR) << __func__ << "Day zero Expt variant is not available";
+    return nullptr;
+  }
+
+  if (day_zero_variant != "a") {
```

Comment:
@simonhong we needed to check for specific variant here. in current expt it's variant 'a'

Reply:
is it only for android? I don't see it from desktop's spec.
In desktop, hiding all or not for feature enabled state.

Reply:
@simonhong we would have the day zero expt always true. We need to check for the variant and if it's 'a' then only we need to apply the changes.

Reply:
got it.

Reply:
It would be great to have a test for this change.